Fix 'You don't have the right permissions to extract archives 
in the folder "file:///path/xds.txtSOME_GARBAGE"' error

--- file-roller-2.32.2/src/fr-window.c.~2~	2015-02-14 01:30:53.114439222 +0300
+++ file-roller-2.32.2/src/fr-window.c	2015-02-14 01:40:48.303433801 +0300
@@ -3946,6 +3960,7 @@
 get_xds_atom_value (GdkDragContext *context)
 {
 	char *ret;
+	gint len;
 
 	g_return_val_if_fail (context != NULL, NULL);
 	g_return_val_if_fail (gdk_drag_context_get_source_window (context) != NULL, NULL);
@@ -3953,9 +3968,12 @@
 	if (gdk_property_get (gdk_drag_context_get_source_window (context),
 			      XDS_ATOM, TEXT_ATOM,
 			      0, MAX_XDS_ATOM_VAL_LEN,
-			      FALSE, NULL, NULL, NULL,
-			      (unsigned char **) &ret))
+			      FALSE, NULL, NULL, &len,
+			      (unsigned char **) &ret)) {
+		ret[len]='\0';
+		debug (DEBUG_INFO, "ret is %s len is %d\n", ret, len);
 		return ret;
+	}
 
 	return NULL;
 }
